### PR TITLE
fix: loro factory impl

### PR DESCRIPTION
**Pull request by zxch3n**

### CODE REVIEW

File: benchmarks/loro/package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
   "author": "Kevin Jahns",
   "license": "MIT",
   "dependencies": {
-    "loro-crdt": "^0.10.0"
+    "loro-crdt": "^0.10.1",
+    "loro-wasm": "^0.10.1"
```
Comment (Leeeon233) [inline benchmarks/loro/package.json:20]: maybe this line can be removed?

Comment (zxch3n) [reply benchmarks/loro/package.json:20]: It's used by measure-bundle. It's fine to keep it here

File: benchmarks/loro/factory.js [hunk 2]
```diff
@@ -11,64 +12,68 @@ export class LoroFactory {
...
+    this.updateHandler = updateHandler;
+    this.list = this.doc.getList('list')
     this.map = this.doc.getMap('map')
     this.text = this.doc.getText('text')
-    this._tr = false
+    this.cachedUpdates = [];
+  }
+
+  update() {
+    if (this.updateHandler) {
```
Comment (dmonad) [inline benchmarks/loro/factory.js:53]: When replaying editing traces, we expect that the crdt implementation generates an update (even if it is not used by the updateHandler). This is also stated in the description of the benchmarks:

https://github.com/dmonad/crdt-benchmarks/blob/7a340cdc385157babf4145de69e7503850ed8300/README.md?plain=1#L52

I made `updateHander` non-optional, which should force all CRDT implementations to send an update to the updateHandler.

File: benchmarks/loro/factory.js [hunk 3]
```diff
@@ -122,50 +124,48 @@ export class LoroCRDT {
...
    */
-  transact (f) {
-    if (this._tr) {
-      f(this, this.doc)
-      return
+  transact(f) {
+    this.cachedUpdates.length = 0;
+    this.inTransact = true;
+    try {
+      f(this)
```
Comment (dmonad) [inline benchmarks/loro/factory.js:146]: I modified this a bit since transactions should also generate updates. They also do so in the other CRDTs.

**Review by dmonad — COMMENTED** [review]
I merged your changes into the `loro` branch. The last loro release made a huge difference!

I made some changes to the factory to make it more consistent with the other factory implementations. I added some comments here as clarifications.

File: benchmarks/loro/factory.js (hunk 2 above)
Comment (zxch3n) [reply benchmarks/loro/factory.js:53]: Got it

File: benchmarks/loro/factory.js (hunk 3 above)
Comment (zxch3n) [reply benchmarks/loro/factory.js:146]: Got it

**Comment by zxch3n** [comment]
> The last loro release made a huge difference!

Probably because the previous release accidentally included a time-consuming checking logic that is used for debugging🫠. https://github.com/loro-dev/loro/commit/d01c03eb779527aa9dc8e9070512ed36ca132e49

**Pull request merged**